spec/requests/contact_spec.rb

```diff
@@ -175,6 +175,23 @@ def anonymous_submission_should_be_successful
     end
   end
 
+  it "should escape user malicious user input when rendering it back to the user" do
+    visit "/contact/govuk"
+    malicious_input = "<script>alert(0)</script>"
+
+    choose "location-0" # The whole site
+    fill_in "Your name", with: malicious_input
+    fill_in "Your email address", with: malicious_input
+    fill_in "textdetails", with: malicious_input
+    click_on "Send message"
+
+    # submission isn't accepted because email address is invalid
+    i_should_be_on "/contact/govuk"
+
+    expect(page.html).to include("&lt;script&gt;alert(0)&lt;/script&gt;")
+    expect(page.html).not_to include(malicious_input)
```

Comment:
```suggestion
    expect(page.html).not_to include(malicious_input)
    
    no_post_calls_should_have_been_made
    expect(page.status_code).to eq(400)
```

Optional: I noticed some of the other tests have these assertions, the first of which arguably has value in that it confirms we're not sending the malicious data elsewhere.